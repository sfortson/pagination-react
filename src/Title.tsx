import React, { useEffect, useState } from 'react';

export interface TitleProps {
  children: string;
  subtitle?: boolean;
  size?: number;
  spaced?: boolean;
}

const Title = (props: TitleProps): JSX.Element => {
  const [titleType, setTitleType] = useState('title');
  const [sizeString, setSizeString] = useState('3');
  const [isSpaced, setIsSpaced] = useState('');

  const { children, size, spaced, subtitle } = props;

  useEffect(() => {
    if (size && size <= 6) {
      setSizeString(size.toString());
    }
  }, [size]);

  useEffect(() => {
    if (subtitle) {
      setTitleType('subtitle');
    }

    if (size && size <= 6) {
      setSizeString(size.toString());
    } else {
      setSizeString('5');
    }
  }, [subtitle, size]);

  useEffect(() => {
    if (spaced) {
      setIsSpaced('is-spaced');
    }
  }, [spaced]);

  const titleClassName =
    isSpaced === 'is-spaced' ? `${titleType} is-${sizeString} ${isSpaced}` : `${titleType} is-${sizeString}`;

  return <p className={titleClassName}>{children}</p>;
};

export default Title;
