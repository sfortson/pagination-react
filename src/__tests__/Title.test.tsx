import React from 'react';
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import Title from '../Title';

let container: HTMLDivElement;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
});

test('Title', () => {
  act(() => {
    ReactDOM.render(<Title>Title</Title>, container);
  });
  expect(container).toMatchSnapshot('Default');

  act(() => {
    ReactDOM.render(<Title size={2}>Title</Title>, container);
  });
  expect(container).toMatchSnapshot('Title size 2');

  act(() => {
    ReactDOM.render(
      <Title size={1} spaced>
        Title
      </Title>,
      container
    );
  });
  expect(container).toMatchSnapshot('Title size 1 and spaced');
});

test('Subtitle', () => {
  act(() => {
    ReactDOM.render(<Title subtitle>Subtitle</Title>, container);
  });
  expect(container).toMatchSnapshot('Default Subtitle');

  act(() => {
    ReactDOM.render(
      <Title size={4} subtitle spaced>
        Subtitle
      </Title>,
      container
    );
  });
  expect(container).toMatchSnapshot('Subtitle size 4 is-spaced');
});
