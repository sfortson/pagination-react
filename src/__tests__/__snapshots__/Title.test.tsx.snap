// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subtitle: Default Subtitle 1`] = `
<div>
  <p
    class="subtitle is-5"
  >
    Subtitle
  </p>
</div>
`;

exports[`Subtitle: Subtitle size 4 is-spaced 1`] = `
<div>
  <p
    class="subtitle is-4 is-spaced"
  >
    Subtitle
  </p>
</div>
`;

exports[`Title: Default 1`] = `
<div>
  <p
    class="title is-5"
  >
    Title
  </p>
</div>
`;

exports[`Title: Title size 1 and spaced 1`] = `
<div>
  <p
    class="title is-1 is-spaced"
  >
    Title
  </p>
</div>
`;

exports[`Title: Title size 2 1`] = `
<div>
  <p
    class="title is-2"
  >
    Title
  </p>
</div>
`;
